Filter

khg0712 opened the pull request.

khg0712 committed:
Add Filters to index.ts

khg0712 committed:
Merge branch 'master' into filter

khg0712 committed:
Add TypeFilter

khg0712 committed:
Modify props interface name

khg0712 committed:
Modify variable name

khg0712 committed:
Modify DropdownItem style

khg0712 committed:
Modify border style, font style

khg0712 committed:
Add DropdownItem hover style

khg0712 committed:
Add FilterDropdown

khg0712 committed:
Add FilterDropdown to TypeFilter

khg0712 committed:
Modify TypeFilter Component to FC

khg0712 committed:
Modify onClickHandler to useCallback

khg0712 committed:
Add DateFilter

engolder commented on src/presentations/DateFilter/index.tsx:
```diff
@@ -0,0 +1,27 @@
+  const isSelected = value !== '날짜';
+  return (
+    <S.DateFilter onClick={onClickFilter}>
+      <S.Dropdownbtn isSelected={isSelected}>{value}</S.Dropdownbtn>
```
`DropdownBtn`은 `DateFilter`와 `TypeFilter`에서 중복으로 사용됨으로 공통컴포넌트로 빼는 것이 좋을 것 같습니다

engolder replied:
`DropdownButton`으로 재명칭하는게 좋을 것 같습니다.
먼저 짧게 줄인 형태인 `btn`보다 이미 `Button`이라는 컴포넌트가 있기 때문에 일관된 명칭으 사용하는 편이 좋을 것 같고,
cabelCase를 사용하는 것이 좋을 것 같습니다

khg0712 replied:
넵

engolder commented on src/presentations/DateFilter/styles.ts:
```diff
@@ -0,0 +1,29 @@
+  &:hover {
+    background: ${({ isSelected }: DropdownbtnProps) =>
+      isSelected ? '#006c70' : '#f2f2f2'};
+    border-color: ${({ isSelected }: DropdownbtnProps) =>
```
함수를 다섯번 사용하는 것 보다는
```ts
${({ isSelected }: DropdownbtnProps) => {
  return css`border-color: ${isSelected ? '#008489' : '#dce0e0'}
    ...
    `
}
```
형태가 가독성에 좋습니다

engolder commented on src/presentations/FilterDropdown/index.ts:
```diff
@@ -0,0 +1,19 @@
+  overflow-x: hidden;
+  max-height: calc(100vh - 152px);
+  background: #fff;
+  ${({ isActive }: FilterDropdownProps) => !isActive && 'display: none;'}
```
`'display: none;'`과 같은 하드코딩보다는
```css`display: none;` ```과 같이 api를 사용하려 해주세요!

engolder commented on src/presentations/TypeFilter/index.tsx:
```diff
@@ -0,0 +1,36 @@
+interface TypeFilterProps {
+  isActive: boolean;
+  value?: string;
+  contentItems: string[];
```
`contentItems: Array<string>`과 같이 작성할 수도 있습니다! (참고)

engolder replied:
`contentItems`보다는 `dropdownItems`가 더 명확한 이름같습니다.

engolder commented on src/presentations/TypeFilter/index.tsx:
```diff
@@ -0,0 +1,36 @@
+  value?: string;
+  contentItems: string[];
+  onSelectType: (value: string) => void;
+  onToggleFilter: (value: boolean) => void;
```
value보다는 isActive나 status같은 명칭이 나을것같네요

engolder replied:
`onToggleFilter`만 보았을때는 무엇을 toggle한다는 것인지 파악하기 힘듭니다.
`onToggleFilterDropdown`과 같이 더 명시적인 네이밍은 어떨까요?

engolder commented on src/presentations/TypeFilter/index.tsx:
```diff
@@ -0,0 +1,36 @@
+  onSelectType,
+  onToggleFilter,
+}) => {
+  const isSelected = value !== '타입';
```
`'타입'`과 같은 리터럴문자열은 오류발생시 추적이어렵습니다.
차라리 default값을 주지말고 value가 undefined인지 아닌지로 selected여부를 판단하는게 좋을 것 같네요

khg0712 replied:
사실 그렇게 짜려고 했는데 머리에서 에러가 났네용

engolder commented on src/presentations/TypeFilter/index.tsx:
```diff
@@ -0,0 +1,36 @@
+  onToggleFilter,
+}) => {
+  const isSelected = value !== '타입';
+  const dropdownItems = contentItems.map((content, i) => (
```
index 인자를 `i`라고 명칭하는 것은 문제가 없지만, 코드간 통일성은 맞춰야할 필요성이 생길 수 있습니다. index변수를 사용하는 상황은 처음이라 수정하실 필요는 없습니다.

engolder commented on src/presentations/TypeFilter/index.tsx:
```diff
@@ -0,0 +1,36 @@
+}) => {
+  const isSelected = value !== '타입';
+  const dropdownItems = contentItems.map((content, i) => (
+    <S.DropdownItem key={i} onClick={onSelectType.bind(null, content)}>
```
단순히 index를 list item에 할당하면 콘솔 에러는 피할 수 있지만, 리액트의 리스트렌더링에 성능적 이점을 놓치게됩니다. 차라리 dropdownItem에 id속성을 만들고 할당하는 편이 좋아보입니다.

engolder replied:
bind에서 첫번째인자로서 undefined대신 null을 사용하신대에는 이유가있나요? 궁금합니다.

engolder committed:
Add dropdown button component

engolder committed:
Add type filter dropdown button component

engolder committed:
Fix hiding shape way css

react-dates component is not work on "display: none" property
so button had to be changed to be have "visibility: hidden" property

engolder committed:
Add date range filter dropdown button component

engolder committed:
Merge pull request #13 from daemabnb/dropdown-button-component

Dropdown button component

engolder committed:
Delete components

engolder committed:
Merge branch 'master' into filter
